Robot_Adapter/CRUD/Update/Elements/Panels.cs

```diff
@@ -0,0 +1,88 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.SurfaceProperties;
+using RobotOM;
+using System.Collections.Generic;
+using BH.oM.Adapter;
+using System.Linq;
+
+namespace BH.Adapter.Robot
+{
+    public partial class RobotAdapter
+    {
+        /***************************************************/
+        /****           Protected Methods               ****/
+        /***************************************************/
+
+        protected bool Update(IEnumerable<Panel> panels)
+        {
+            Dictionary<int, HashSet<string>> panelTags = GetTypeTags(typeof(Panel));
+            RobotObjObjectServer robotObjectServer = m_RobotApplication.Project.Structure.Objects;
+            foreach (Panel panel in panels)
+            {
+                RobotObjObject robotPanel = robotObjectServer.Get((int)panel.CustomData[AdapterIdName]) as RobotObjObject;
+                if (robotPanel == null)
+                    return false;
```

Comment:
Might tweak the beginning part here, down to the rotpanel == null check to something like:

```
                if (panel == null)
                {
                    Engine.Reflection.Compute.RecordWarning("At least one provided panel is null and is not updated!");
                    continue;
                }

                if (!panel.CustomData.ContainsKey(AdapterIdName))
                {
                    string panelName = string.IsNullOrWhiteSpace(panel.Name) ? "no name" : "name " + panel.Name;
                    Engine.Reflection.Compute.RecordWarning("Panel with " + panelName + " did not contain any Robot id. To update panels they need this information. For this operation to work, try using a Panel that has first been pulled from Robot");
                    continue;
                }

                RobotObjObject robotPanel = robotObjectServer.Get((int)panel.CustomData[AdapterIdName]) as RobotObjObject;
                if (robotPanel == null)
                {
                    Engine.Reflection.Compute.RecordWarning("Could not find a panel with the Id " + panel.CustomData[AdapterIdName].ToString() + " in Robot. Panel could not be updated!");
                    continue;
                }
```